packages/components/src/mobile/bottom-sheet/picker-cell.native.js

```diff
@@ -23,7 +18,7 @@ export default function BottomSheetPickerCell( props ) {
 		onChangeValue( newValue );
 	};
 
-	const option = find( options, { value } );
+	const option = options.find( ( opt ) => opt.value === value );
```

Comment:
`options` will default to an array:

https://github.com/WordPress/gutenberg/blob/1d778aa5e7506390c4d1a89974ec69088026855a/packages/components/src/select-control/index.native.js#L17